DCOS-8071: Fix status dots in task detail

Poltergeist opened the pull request:
This adds the right coloring for task states:

![screen shot 2016-07-05 at 17 22 33](https://cloud.githubusercontent.com/assets/156010/16620958/c01b1b58-4394-11e6-9770-4d0a3775d657.png)
![screen shot 2016-07-05 at 17 22 07](https://cloud.githubusercontent.com/assets/156010/16620957/c01a6ff0-4394-11e6-92ac-6266ca5f03fa.png)

mesosphere-web commented:
Reviewer: @orlandohohmeier

jfurrow commented:
This looks great! Nice work!

daiyiwhale commented:
It's beautiful (': ✨

Poltergeist commented:
By the way are all states right?

\cc @leemunroe

orlandohohmeier commented on src/js/components/TaskTable.js:
```diff
@@ -157,9 +143,40 @@ class TaskTable extends React.Component {
+
+    let running = ['TASK_RUNNING', 'TASK_STARTING'].includes(state) && unknown;
+    let success = healthy && state === 'TASK_RUNNING';
+    let danger = (dangerState && !activeState && dangerState &&
```
Let's check `dangerState` only once... 😀

orlandohohmeier replied:
Any chance we can refactor this into separate sequential conditions? Might be easier to read.

orlandohohmeier commented:
Let's please double check it this state is valid:
![](https://cloud.githubusercontent.com/assets/647035/16649477/672d6d2a-4439-11e6-97bb-cef92a5e671b.png)

leemunroe commented:
Looking great. Couple of nitpicks, open to discussion if you disagree on any.

* Looks like there are 2 shades of grey text going on e.g. `Finished` is different than `Killed`. Is there a good reason? I'd expect them to be the same shade.
* I'm not quite sure what the sequence is of all these statuses occur, can you confirm? `Created` > `Starting` > `Started` > `Running`
* Now that I see `Staging` as yellow I'm wondering why we decided that. I think we copied that from Marathon. Does it still make sense to call that status out as yellow?

Poltergeist commented:
The 2 shades of grey came from a document in which light grey was mentioned for those states. I can change that. That is also were the yellow color came from for `Staging`. 

I am currently trying to confirm that the status sequence is right.

leemunroe commented:
* There are two shades of grey for the dots, but the text should be the one shade of grey so it is legible
* And you're right about the yellow for staging, I just wanted to check with you guys if you had an opinion, lgtm

Poltergeist commented:
![image](https://cloud.githubusercontent.com/assets/156010/16660584/48ecb15a-446f-11e6-86ee-52f4aa3a8b34.png)
Updated the styles and removed the 2 shades of grey in the fonts.

The yellow is maybe drawing to much attention, staging isn't a bad state right? As a user I might be confused? I always consider red to be the color of an error and yellow being a warning / notice of unusual behaviour. This is what I do think about it.

kennyt commented:
hmm i'm currently testing it and staging looks like this:

![px](https://cl.ly/3g1i3p0i1x3u/Image%202016-07-07%20at%2010.15.29%20AM.png)

leemunroe commented:
Yeh seems odd to me. Lets make it white. We can change it in the future if we find there's a good reason to call it out.

Re. staging having a red dot, can a task be unhealthy before it is `Running`? Doesn't seem right to me.

Poltergeist commented:
All states documented in Mesos in their order:

```
  TASK_STAGING
  TASK_STARTING
  TASK_RUNNING 

  TASK_KILLING

  TASK_FINISHED
  TASK_FAILED
  TASK_KILLED
  TASK_LOST
  TASK_ERROR 
```
I'll check for the staging part.

orlandohohmeier commented:
Is there a reason for not coloring tasks of suspended/idle services?

![screen shot 2016-07-08 at 15 47 42](https://cloud.githubusercontent.com/assets/647035/16690785/d773dbc8-4529-11e6-97a3-972e4ff5c467.png)
![screen shot 2016-07-08 at 15 47 53](https://cloud.githubusercontent.com/assets/647035/16690784/d77248ee-4529-11e6-8c4b-3a97ce6be448.png)

Poltergeist commented:
This doesn't seem right...

Poltergeist commented:
**mesosphere-ci** retest this please

pierlo-upitup commented:
@orlandohohmeier i cant reproduce your 2nd screenshot. Can you ?

orlandohohmeier commented:
LGTM 💯 ✨
